cf/data/data.py

```diff
@@ -10956,9 +10956,10 @@ def flip(self, axes=None, inplace=False, i=False):
         if not iaxes:
             return d
 
-        index = [
-            slice(None, None, -1) if i in axes else slice(None) for i in iaxes
```

Comment:
@davidhassell I believe this present list comprehension is equivalent to:

```python
if i in iaxes:
    index = [slice(None, None, -1) for i in iaxes]
else:
    index = [slice(None) for i in iaxes]
```

whereas what was intended is as suggested in this PR, varying only in the second line (but then becoming contrived and potentially confusing as a list comprehension, hence converting it to `if`/`else` is better), which makes the original `test_Data_flip` unit test pass where it would otherwise fail on the `self.assertTrue((d.array == array).all())`. Please do let me know if that sounds about right to you...

Reply:
Ignore the above, it makes no sense whatsoever, but for some reason the PR code makes the unit test work. I am very confused...